services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -0,0 +1,213 @@
+            } else {
+                flowPairBuilder.reverse(flow);
+            }
+        }
+        return flowPairBuilder.build();
+    }
+
+    /**
+     * Get all flows grouped in FlowPairs.
+     * @return List of flow pairs
+     */
+    public List<FlowPair> getFlows() {
+        Collection<Flow> flows = flowRepository.findAll();
+        Map<String, FlowPairBuilder> flowPairs = new HashMap<>();
+        for (Flow f : flows) {
+            String flowId = f.getFlowId();
+            FlowPair.FlowPairBuilder flowPairBuilder;
+            if (flowPairs.containsKey(flowId)) {
+                flowPairBuilder = flowPairs.get(flowId);
+            } else {
+                flowPairBuilder = FlowPair.builder();
+                flowPairs.put(flowId, flowPairBuilder);
+            }
+            if (f.isForward()) {
+                flowPairBuilder.forward(f);
+            } else {
+                flowPairBuilder.reverse(f);
+            }
+
+        }
+        List<FlowPair> flowPairList = new ArrayList<>();
+        for (FlowPairBuilder builder :flowPairs.values()) {
+            flowPairList.add(builder.build());
+        }
+        return flowPairList;
+    }
+
+    /**
+     * Update status for selected flow.
+     * @param flowId - target flow to update
+     * @param flowStatus - new status
+     * @return target FlowPair
+     */
+    public FlowPair updateFlowStatus(String flowId, FlowStatus flowStatus) {
+        TransactionManager transactionManager = this.persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            FlowPair pair = getFlowPair(flowId);
+            Flow forward = pair.getForward();
+            Flow reverse = pair.getReverse();
+            if (forward != null && reverse != null) {
+                forward.setStatus(flowStatus);
+                flowRepository.createOrUpdate(forward);
+
+                reverse.setStatus(flowStatus);
+                flowRepository.createOrUpdate(reverse);
+            }
+            transactionManager.commit();
+            return pair;
+        } catch (Exception e) {
+            transactionManager.rollback();
+            throw e;
+        }
+    }
+
+    private void processDeleteFlow(Flow flow) {
+        flowSegmentRepository.deleteFlowSegments(flow);
+        flowRepository.delete(flow);
+        islRepository.updateIslBandwidth(flow);
+    }
+
+    /**
+     * Replace existing flow in DB with the new one with similar id and updated info.
+     *
+     * @param flowPair - forward and reverse pairs to be processed
+     */
+    public void updateFlow(FlowPair flowPair) {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            Flow flow = flowPair.getForward();
+            flowSegmentRepository.deleteFlowSegments(flow);
+            flowRepository.deleteByFlowId(flow.getFlowId());
+            createFlowForPair(flowPair);
```

Comment:
Can we have "proper" update operation (not delete+create)? It will be required for object's history tracking.

PS Where is the bandwidth recalculation?

Reply:
+1 no need to delete/recreate the flow instead of update.